web/shared/src/typing_status.ts

```diff
@@ -1,5 +1,17 @@
 import _ from "lodash";
 
+type TypingStatusWorker = {
+    get_current_time: () => number;
+    notify_server_start: (recipient: number[] | null) => void;
+    notify_server_stop: (recipient: number[] | null) => void;
```

Comment:
Do these notify functions actually accept `null`? I don't see how they could do anything coherent with `null` passed in, and I would have thought that all the callers would ensure the recipient is not `null` in advance.

Reply:
Yeah that is true, these functions never accept `null` parameter. Fixed this!